### PR TITLE
implement default settings

### LINKED ISSUE #6: Implement and provide default settings
So that projects don't need to provide/duplicate all the settings and we ship with sane defaults.

For example: https://github.com/jazzband/django-cookie-consent/blob/master/cookie_consent/conf.py

### PR DESCRIPTION
**Pull request by sergei-maertens**
Closes #6

### CODE REVIEW

**Review by vaszig — APPROVED** [review]

File: docs/quickstart.rst [hunk 1]
```diff
@@ -21,66 +22,89 @@ Installation
...
 
-#.  Update your ``settings.py`` file with the following (if you haven't defined 
-    logging yet, you can see the Django's `documentation`_):
+#. Run ``python manage.py migrate`` to create the necessary database tables
+
+Configuration
+=============
+
+You must configure the ``LOGGING`` Django setting to integrate the formatter(s) and/or
+handler(s). Review the Django's `documentation`_ about this setting if there is no
```
Comment (pi-sigma) [inline docs/quickstart.rst:31]: Either *Review the Django `documentation`* or *Review Django's `documentation`*

File: docs/quickstart.rst [hunk 2]
```diff
@@ -21,66 +22,89 @@ Installation
...
-                "application/json",
-                "application/xml",
-                "application/soap+xml",
-        ] # save request/response bodies with matching content type
-        LOG_OUTGOING_REQUESTS_MAX_CONTENT_LENGTH = 524_288  # maximal size (in bytes) for the request/response body
-        LOG_OUTGOING_REQUESTS_LOG_BODY_TO_STDOUT = True
 
+The library ships with safe defaults for settings - essentially only emitting
+meta-information about requests and responses. To view request and response bodies,
+you likely want to apply the following non-default settings:
```
Comment (pi-sigma) [inline docs/quickstart.rst:76]: This makes it sound like it's entirely optional to add the following settings. But is that right? the `save_logs_enabled` and `save_body_enabled` properties of `OutgoingRequestsLogConfig` check if the admin setting for saving is `use_default`. If it is, they access `settings.LOG_OUTGOING_REQUESTS_DB_SAVE` and `settings.LOG_OUTGOING_REQUESTS_DB_SAVE_BODY` respectively. So even if they don't want to save logs, they have to specify this explicitly in the settings.

File: log_outgoing_requests/formatters.py [hunk 3]
```diff
@@ -1,14 +1,30 @@
 import logging
 import textwrap
 
-from django.conf import settings
-
 
 class HttpFormatter(logging.Formatter):
+    """
+    Display request and response (meta) details python requests library log records.
```
Comment (pi-sigma) [inline log_outgoing_requests/formatters.py:7]: ...details of python requests...

File: docs/quickstart.rst [hunk 4]
```diff
@@ -21,66 +22,89 @@ Installation
...
 
-#.  Run ``python manage.py migrate`` to create the necessary database tables.
+.. code-block:: python
+
+    LOG_OUTGOING_REQUESTS_DB_SAVE = True # save logs enabled/disabled based on the boolean value
+    LOG_OUTGOING_REQUESTS_DB_SAVE_BODY = True # save request/response body
+    LOG_OUTGOING_REQUESTS_EMIT_BODY = True # log request/response body
+
+.. note:: Saving to database (both the entire record and the request/response bodies)
```
Comment (pi-sigma) [inline docs/quickstart.rst:84]: Should we mention the possibility of specifying max size here? 

In addition, you can specify the maximum size (in bytes) for contents to be saved, like so:

LOG_OUTGOING_REQUESTS_MAX_CONTENT_LENGTH = 524_288 # 0.5 MB

File: docs/quickstart.rst [hunk 5]
```diff
@@ -21,66 +22,89 @@ Installation
...
-                "text/*",
-                "application/json",
-                "application/xml",
-                "application/soap+xml",
-        ] # save request/response bodies with matching content type
-        LOG_OUTGOING_REQUESTS_MAX_CONTENT_LENGTH = 524_288  # maximal size (in bytes) for the request/response body
-        LOG_OUTGOING_REQUESTS_LOG_BODY_TO_STDOUT = True
 
+The library ships with safe defaults for settings - essentially only emitting
+meta-information about requests and responses. To view request and response bodies,
```
Comment (pi-sigma) [inline docs/quickstart.rst:75]: Perhaps mention where the settings are stored ("...ships with sane defaults for settings, accessible via conf.py - essentially...")?

**Review by pi-sigma — COMMENTED** [review]
Looking good, just a few minor questions/suggestions.

File: docs/quickstart.rst (hunk 4 above)
Comment (sergei-maertens) [reply docs/quickstart.rst:84]: I'm deliberately leaving out the setting here as I believe the default of 0.5 MB should just work. The other remarks are now incorporated!

File: docs/quickstart.rst (hunk 5 above)
Comment (sergei-maertens) [reply docs/quickstart.rst:75]: I don't think this is needed, as how we consume the settings in the library is our implementation detail. Users of the library only need to know what they can override.

**Pull request merged**